### PR TITLE
TST: add tests for period resample sum with min_count

### LINKED ISSUE #19974: min_count is ignored by sum/prod when resampling a PeriodIndex
#### Code Sample
Setup:
```python
import numpy as np
import pandas as pd
​
index = pd.date_range(start='2018', freq='M', periods=6)
data = np.ones(6)
data[3:6] = np.nan

datetime = pd.Series(data, index)
period = datetime.to_period()
```

Resampling and summing (`min_count=1`) the Series with a DatetimeIndex:

```python
datetime
datetime.resample('Q').sum(min_count=1)
```
```
2018-01-31   1.0
2018-02-28   1.0
2018-03-31   1.0
2018-04-30   NaN
2018-05-31   NaN
2018-06-30   NaN
Freq: M, dtype: float64
```
```
2018-03-31    3.0
2018-06-30    NaN
Freq: Q-DEC, dtype: float64
```

Resampling and summing (`min_count=1`) the Series with a PeriodIndex:

```python
period
period.resample('Q').sum(min_count=1)
```
```
2018-01    1.0
2018-02    1.0
2018-03    1.0
2018-04    NaN
2018-05    NaN
2018-06    NaN
Freq: M, dtype: float64
```
```
2018Q1    3.0
2018Q2    0.0
Freq: Q-DEC, dtype: float64
```

#### Problem description
`sum()` and `prod()` seem to ignore the `min_count` argument when used on a resampled series or dataframe with a PeriodIndex.

#### Expected Output
I would expect the same result whether using a DatetimeIndex or PeriodIndex. Specifically, in the example above, 2018Q2 should be NaN.

#### Output of ``pd.show_versions()``
<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.4.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 61 Stepping 4, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.22.0
pytest: 3.2.1
pip: 9.0.1
setuptools: 36.5.0.post20170921
Cython: 0.26.1
numpy: 1.13.3
scipy: 0.19.1
pyarrow: None
xarray: None
IPython: 6.1.0
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.2
feather: None
matplotlib: 2.1.0
openpyxl: 2.4.8
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml: 4.1.0
bs4: 4.6.0
html5lib: 0.999999999
sqlalchemy: 1.1.13
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by ganevgv**
- [x] closes #19974
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/resample/test_datetime_index.py [hunk 1]
```diff
@@ -694,6 +694,20 @@ def test_resample_timestamp_to_period(simple_date_range_series):
     tm.assert_series_equal(result, expected)
 
 
+def test_datetime_resample_sum_min_count():
```
Comment (jreback) [inline pandas/tests/resample/test_datetime_index.py]: can you move this to the base class where its tests all of DTI, PI, TDI; pls use the fixtures.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/resample/test_datetime_index.py [hunk 2]
```diff
@@ -694,6 +694,20 @@ def test_resample_timestamp_to_period(simple_date_range_series):
...
+    # GH 19974
+    index = date_range(start="2018", freq="M", periods=6)
```
Comment (jreback) [inline pandas/tests/resample/test_datetime_index.py]: this is not actually testing for the issue as this is a DTI and not a PI.

Comment (ganevgv) [reply pandas/tests/resample/test_datetime_index.py]: Yes, this test was using DTI as that was referenced as the expected behaviour in the issue.
Not it's removed and the only test left is using PI - `test_period_resample_sum_min_count` in `pandas/tests/resample/test_period_index.py`.

File: pandas/tests/resample/test_period_index.py [hunk 3]
```diff
@@ -548,6 +548,19 @@ def test_quarterly_resampling(self):
         exp = ts.to_timestamp().resample("A").mean().to_period()
         tm.assert_series_equal(result, exp)
 
+    def test_period_resample_sum_min_count(self):
```
Comment (jreback) [inline pandas/tests/resample/test_period_index.py]: can this test be done more generally in test_base?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/resample/test_period_index.py (hunk 3 above)
Comment (ganevgv) [reply pandas/tests/resample/test_period_index.py]: thanks for the feedback @jreback - moved the test and generalised to include tests for DTI, PI and TDI.

File: pandas/tests/resample/test_base.py [hunk 4]
```diff
@@ -274,3 +274,79 @@ def test_resample_quantile(series):
...
+    ],
+)
+def test_resample_sum(
+    series, freq, result_name, result_data, result_index, result_freq
+):
+    # GH 19974
+    series[:5] = 1
+    series[5:] = np.nan
+
+    if isinstance(series.index, TimedeltaIndex) and freq != "D":
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: put this particular case in the test_timedelta.py as its only relevant there and makes the test setup way too complicated.

File: pandas/tests/resample/test_base.py [hunk 5]
```diff
@@ -274,3 +274,79 @@ def test_resample_quantile(series):
...
+        msg = ".* is a non-fixed frequency"
+        with pytest.raises(ValueError, match=msg):
+            result = series.resample(freq).sum(min_count=1)
+
+    else:
+        result = series.resample(freq).sum(min_count=1)
+
+        if isinstance(series.index, DatetimeIndex) and result_name == "dti":
+            index = DatetimeIndex(result_index, freq=result_freq)
+            expected = Series(result_data, index, name=result_name)
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: you are duplicating a lot of code here, pls share

File: pandas/tests/resample/test_base.py [hunk 6]
```diff
@@ -274,3 +274,79 @@ def test_resample_quantile(series):
...
+            tm.assert_series_equal(result, expected)
+
+        if isinstance(series.index, PeriodIndex) and result_name == "pi":
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: instead of doing this you can simply add a 'box' argument that does the appropriate conversion in the cases

e.g.

for 'pi'

PeriodIndex

and so on.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/resample/test_base.py (hunk 4 above)
Comment (ganevgv) [reply pandas/tests/resample/test_base.py]: makes sense - moved the tdi test to `test_timedelta.py`

File: pandas/tests/resample/test_base.py (hunk 6 above)
Comment (ganevgv) [reply pandas/tests/resample/test_base.py]: thanks for the feedback @jreback - changed the test so that there's no duplication anymore

File: pandas/tests/resample/test_base.py [hunk 7]
```diff
@@ -274,3 +274,72 @@ def test_resample_quantile(series):
     result = s.resample(freq).quantile(q)
     expected = s.resample(freq).agg(lambda x: x.quantile(q)).rename(s.name)
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "_index_factory,_series_name,_index_start,_index_end", [DATE_RANGE, PERIOD_RANGE]
+)
+@pytest.mark.parametrize(
```
Comment (jreback) [inline pandas/tests/resample/test_base.py]: @ganevgv rather than nesting a dict, just make data, index paramters e.g.

```
@pytest.mark.parameterize('freq, data, index')
```

otherwise this is very hard to follow.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/resample/test_base.py (hunk 7 above)
Comment (ganevgv) [reply pandas/tests/resample/test_base.py]: @jreback - removed the nested dict and changed the format to: 
```
@pytest.mark.parameterize("freq, result_name, result_data, result_index, result_freq")
```

**Comment by jreback** [comment]
can you merge master and will look again

**Comment by ganevgv** [comment]
@jreback - rebased onto latest master.

**Comment by alimcmaster1** [comment]
@ganevgv - sorry can you merge master again - there are some conflicts

**Comment by simonjayhawkins** [comment]
superseded by #31196
